DependencyInjection/CompilerPass/ApiPass.php

```diff
@@ -5,13 +5,32 @@
 use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
 use Symfony\Component\DependencyInjection\ContainerBuilder;
 use Symfony\Component\DependencyInjection\Reference;
+use SymfonyCorp\Connect\Api\Api;
 
 class ApiPass implements CompilerPassInterface
 {
     public function process(ContainerBuilder $container)
     {
-        if ($container->getParameter('kernel.debug')) {
-            $container->getDefinition('symfony_connect.buzz')->addMethodCall('addListener', array(new Reference('symfony_connect.collector.api')));
+        if ((new \ReflectionClass(Api::class))->hasMethod('getStringResponse')) {
+            // connect ~5.0
+            if ($container->getParameter('kernel.debug')) {
+                $container->getDefinition('symfony_connect.buzz')->addMethodCall('addListener', array(new Reference('symfony_connect.collector.api')));
+            }
+        } else {
+            // connect ^5.1
+            if ($container->hasDefinition('http_client')) {
+                $container->getDefinition('symfony_connect.api')->replaceArgument(1, new Reference('http_client'));
+                $container->getDefinition('symfony_connect.oauth_consumer')->replaceArgument(4, new Reference('http_client'));
+            }
+
+            $container->getDefinition('symfony_connect.collector.api')->clearTags();
```

Comment:
should be inside the above "if", right?

Reply:
no? in this case `connect` 5.1 is being used, so i don't think we should collect anything